src/main/kotlin/no/nav/personbruker/dittnav/eventaggregator/metrics/ProducerNameResolver.kt

```diff
@@ -0,0 +1,46 @@
+package no.nav.personbruker.dittnav.eventaggregator.metrics
+
+import kotlinx.coroutines.runBlocking
+import no.nav.personbruker.dittnav.eventaggregator.common.database.Database
+import no.nav.personbruker.dittnav.eventaggregator.metrics.db.getProdusentnavn
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.lang.Exception
+import java.time.Duration
+import java.time.LocalDateTime
+
+class ProducerNameResolver(private val database: Database) {
+
+    private var producerNameAliases: Map<String, String> = emptyMap()
+    private var lastRetrievedFromDB: LocalDateTime? = null
+    private val PRODUCERNAME_CACHE_IN_MINUTES = 15
+
+    private val log: Logger = LoggerFactory.getLogger(ProducerNameResolver::class.java)
+
+    fun getProducerNameAliases(): Map<String, String> {
+        return runBlocking {
```

Comment:
Kunne withContext blitt brukt i stede for å ikke blokkere? F.eks. withContext(Dispatchers.IO).

Reply:
Ja, det tror jeg absolutt at det kan. Skal endre.